src/App.jsx

```diff
@@ -1,28 +1,91 @@
-import React, { useState } from 'react';
+import React, { useState, useRef, useEffect } from 'react';
 import { DragDropContext, Droppable, Draggable } from 'react-beautiful-dnd';
+import EmojiPicker from 'emoji-picker-react';
 import './App.css';
 
 const initialColumns = {
   todo: {
     id: 'todo',
     title: 'To Do',
-    tasks: []
-  },
-  inProgress: {
-    id: 'inProgress',
-    title: 'In Progress',
-    tasks: []
-  },
-  done: {
-    id: 'done',
-    title: 'Done',
-    tasks: []
+    tasks: [],
+    order: 0,
+    emoji: '📝',
+    color: '#1a73e8'
   }
 };
 
 function App() {
   const [columns, setColumns] = useState(initialColumns);
   const [newTask, setNewTask] = useState('');
+  const [newColumnTitle, setNewColumnTitle] = useState('');
```

Comment:
![kody code-review](https://img.shields.io/badge/kody-code--review-brightgreen)

```suggestion
if (!newColumnTitle.trim()) return;
if (Object.values(columns).some(col => col.title === newColumnTitle.trim())) {
  alert('A column with this title already exists');
  return;
}
```
Add validation to prevent duplicate column titles and handle potential collisions in columnId generation.



<sub>Talk to Kody by mentioning @kody</sub>

<sub>Was this suggestion helpful? React with 👍 or 👎 to help Kody learn from this interaction.</sub><!-- kody-codereview -->&#8203;
&#8203;